src/gui/Src/Gui/BreakpointsView.cpp

```diff
@@ -660,6 +662,97 @@ void BreakpointsView::editBreakpointSlot()
     }
 }
 
+void BreakpointsView::editBreakpointsSlots()
+{
+    EditBreakpointsDialog dialog(this, "Edit Selected BreakPoints");
+
+    if(dialog.exec() != QDialog::Accepted)
+        return;
+
+    auto exec = [](const QString & command)
+    {
+        DbgCmdExecDirect(command.toUtf8().constData());
+    };
+
+    const BRIDGEBP & newBp = dialog.getBp();
+
+    for(int i : getSelection())
+    {
+        if(!isValidBp(i))
+            continue;
+
+        auto & bp = selectedBp(i);
+
+        QString addrText = QString().sprintf("\"%s\":$%X", bp.mod, bp.addr);
+
+        switch(bp.type)
+        {
+        case bp_normal:
+            exec(QString("SetBreakpointName %1, \"%2\"").arg(addrText).arg(newBp.name));
+            exec(QString("SetBreakpointCondition %1, \"%2\"").arg(addrText).arg(newBp.breakCondition));
+            exec(QString("SetBreakpointLog %1, \"%2\"").arg(addrText).arg(newBp.logText));
+            exec(QString("SetBreakpointLogCondition %1, \"%2\"").arg(addrText).arg(newBp.logCondition));
+            exec(QString("SetBreakpointCommand %1, \"%2\"").arg(addrText).arg(newBp.commandText));
+            exec(QString("SetBreakpointCommandCondition %1, \"%2\"").arg(addrText).arg(newBp.commandCondition));
+            exec(QString("ResetBreakpointHitCount %1, %2").arg(addrText).arg(ToPtrString(newBp.hitCount)));
+            exec(QString("SetBreakpointFastResume %1, %2").arg(addrText).arg(newBp.fastResume));
+            exec(QString("SetBreakpointSilent %1, %2").arg(addrText).arg(newBp.silent));
+            exec(QString("SetBreakpointSingleshoot %1, %2").arg(addrText).arg(newBp.singleshoot));
+            break;
+        case bp_hardware:
+            exec(QString("SetHardwareBreakpointName %1, \"%2\"").arg(addrText).arg(newBp.name));
+            exec(QString("SetHardwareBreakpointCondition %1, \"%2\"").arg(addrText).arg(newBp.breakCondition));
+            exec(QString("SetHardwareBreakpointLog %1, \"%2\"").arg(addrText).arg(newBp.logText));
+            exec(QString("SetHardwareBreakpointLogCondition %1, \"%2\"").arg(addrText).arg(newBp.logCondition));
+            exec(QString("SetHardwareBreakpointCommand %1, \"%2\"").arg(addrText).arg(newBp.commandText));
+            exec(QString("SetHardwareBreakpointCommandCondition %1, \"%2\"").arg(addrText).arg(newBp.commandCondition));
+            exec(QString("ResetHardwareBreakpointHitCount %1, %2").arg(addrText).arg(ToPtrString(newBp.hitCount)));
+            exec(QString("SetHardwareBreakpointFastResume %1, %2").arg(addrText).arg(newBp.fastResume));
+            exec(QString("SetHardwareBreakpointSilent %1, %2").arg(addrText).arg(newBp.silent));
+            exec(QString("SetHardwareBreakpointSingleshoot %1, %2").arg(addrText).arg(newBp.singleshoot));
+        case bp_memory:

```

Comment:
case `bp_hardware` falls through to `bp_memory` due to a lack of a break - I'm not sure about the specifics of setting hardware breakpoints so i'm not sure if this is or is not intended